eclipse-collections/src/main/java/com/fasterxml/jackson/datatype/eclipsecollections/EclipseCollectionsTypeModifier.java

```diff
@@ -0,0 +1,22 @@
+package com.fasterxml.jackson.datatype.eclipsecollections;
+
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.type.CollectionLikeType;
+import com.fasterxml.jackson.databind.type.TypeBindings;
+import com.fasterxml.jackson.databind.type.TypeFactory;
+import com.fasterxml.jackson.databind.type.TypeModifier;
+import java.lang.reflect.Type;
+import org.eclipse.collections.api.collection.ImmutableCollection;
+
+class EclipseCollectionsTypeModifier extends TypeModifier {
+    @Override
+    public JavaType modifyType(JavaType type, Type jdkType, TypeBindings context, TypeFactory typeFactory) {
+        if (!type.isCollectionLikeType()) {
+            JavaType collectionType = type.findSuperType(ImmutableCollection.class);
```

Comment:
is this a hot enough path to justify an isAssignableFrom check or is this fine?

Reply:
Which check are you referring to? `isCollectionLikeType()` is cheap so I think that is fine.
And `findSuperType()` is necessary to find type parameterization.

Reply:
Oh. Ah, now I get it; to see if `findSuperType()` is needed. Hmmh. Ideally it should not be on hot path... but I probably would have added guard statement just in case. Should be fine either way.